tests/NHSD.BuyingCatalogue.Organisations.Api.UnitTests/Controllers/OrganisationControllerTests.cs

```diff
@@ -263,11 +284,61 @@ public async Task UpdateOrganisationByIdAsync_OrganisationRepository_UpdateAsync
         }
 
         [Test]
-        public void UpdateOrganisationByIdAsync_NullUpdateViewModel_ThrowsException()
+        public void UpdateOrganisationByIdAsync_NullUpdateViewModel_Throws()
         {
-            using var controller = OrganisationControllerBuilder.Create().WithUpdateOrganisation(new Organisation()).Build();
+            Assert.ThrowsAsync<ArgumentNullException>(async () =>
+            {
+                using var controller = OrganisationControllerBuilder.Create().WithUpdateOrganisation(new Organisation()).Build();
+                await controller.UpdateOrganisationByIdAsync(Guid.Empty, null);
+            });
+        }
+
+        [Test]
+        public async Task CreateOrganisationAsync_ServiceReturnsSuccess_Returns_Created()
+        {
+            var organisationId = Guid.NewGuid();
+            using var controller = OrganisationControllerBuilder.Create().WithCreateOrganisation().WithCreateOrganisationService(true, organisationId.ToString()).Build();
+
+            var response = await controller.CreateOrganisationAsync(new CreateOrganisationRequestViewModel());
+
+            response.Should().BeOfType<ActionResult<CreateOrganisationResponseViewModel>>();
 
-            Assert.ThrowsAsync<ArgumentNullException>(async () => await controller.UpdateOrganisationByIdAsync(Guid.Empty, null));
+            var expected = new CreatedResult(new Uri($"/{organisationId}", UriKind.Relative),
+                new CreateOrganisationResponseViewModel { OrganisationId = organisationId.ToString() });
+
+            var actual = response.Result;
+
+            actual.Should().BeEquivalentTo(expected);
+        }
+
+        [Test]
+        public async Task CreateOrganisationAsync_ServiceReturnsFailure_Returns_BadRequest()
+        {
+            const string errorMessage = "Some Error Message Id";
+            using var controller = OrganisationControllerBuilder.Create().WithCreateOrganisation().WithCreateOrganisationService(false, errorMessage).Build();
+
+            var response = await controller.CreateOrganisationAsync(new CreateOrganisationRequestViewModel());
+
+            response.Should().BeOfType<ActionResult<CreateOrganisationResponseViewModel>>();
+
+            var expected = new BadRequestObjectResult(new CreateOrganisationResponseViewModel
+            {
+                Errors = new [] { new ErrorMessageViewModel(errorMessage) }
+            });
+
+            var actual = response.Result;
+
+            actual.Should().BeEquivalentTo(expected);
+        }
+
+        [Test]
+        public void CreateOrganisationAsync_NullCreateOrganisationRequestViewModel_Throws()
+        {
+            Assert.ThrowsAsync<ArgumentNullException>(async () =>
+            {
+                using var controller = OrganisationControllerBuilder.Create().WithCreateOrganisation().Build();
```

Comment:
Is `WithCreateOrganisation()` required here?

Reply:
👍